src/storage/redis_db.cc

```diff
@@ -84,25 +84,25 @@ rocksdb::Status Database::ParseMetadata(RedisTypes types, Slice *bytes, Metadata
   return s;
 }
 
-rocksdb::Status Database::GetMetadata(RedisTypes types, const Slice &ns_key, Metadata *metadata) {
+rocksdb::Status Database::GetMetadata(GetOptions options, RedisTypes types, const Slice &ns_key, Metadata *metadata) {
   std::string raw_value;
   Slice rest;
-  return GetMetadata(types, ns_key, &raw_value, metadata, &rest);
+  return GetMetadata(options, types, ns_key, &raw_value, metadata, &rest);
 }
 
-rocksdb::Status Database::GetMetadata(RedisTypes types, const Slice &ns_key, std::string *raw_value, Metadata *metadata,
+rocksdb::Status Database::GetMetadata(GetOptions options, RedisTypes types, const Slice &ns_key, std::string *raw_value, Metadata *metadata,
                                       Slice *rest) {
-  auto s = GetRawMetadata(ns_key, raw_value);
+  auto s = GetRawMetadata(options, ns_key, raw_value);
   *rest = *raw_value;
   if (!s.ok()) return s;
   return ParseMetadata(types, rest, metadata);
 }
 
-rocksdb::Status Database::GetRawMetadata(const Slice &ns_key, std::string *bytes) {
-  LatestSnapShot ss(storage_);
-  rocksdb::ReadOptions read_options;
-  read_options.snapshot = ss.GetSnapShot();
-  return storage_->Get(read_options, metadata_cf_handle_, ns_key, bytes);
+rocksdb::Status Database::GetRawMetadata(GetOptions options, const Slice &ns_key, std::string *bytes) {
+  rocksdb::ReadOptions opts;
```

Comment:
Don't know why we get a latest snapshot previously. Is this by design?

Reply:
That's a good question. We obtained a snapshot, got only the metadata from it and then threw it away.

Reply:
Yeah, @git-hulk is that by design (related to txn) or just writing that there?

Reply:
I cannot quite remember the detail, and we discussed using the same snapshot in the same command(transaction). But it failed since many places need to be modified, I guess that's why some places are explicitly using the snapshot but some of them NOT.

Reply:
Nice. @git-hulk Here I allow get without a snapshot in rocksdb, I guess it might maintaining the same syntax and making read a little bit faster